metricbeat/module/rabbitmq/exchange/data.go

```diff
@@ -0,0 +1,96 @@
+package exchange
+
+import (
+	"encoding/json"
+
+	"github.com/elastic/beats/libbeat/common"
+	s "github.com/elastic/beats/libbeat/common/schema"
+	c "github.com/elastic/beats/libbeat/common/schema/mapstriface"
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+var (
+	schema = s.Schema{
+		"name":        c.Str("name"),
+		"vhost":       c.Str("vhost"),
+		"type":        c.Str("type"),
+		"durable":     c.Bool("durable"),
+		"auto_delete": c.Bool("auto_delete"),
+		"internal":    c.Bool("internal"),
+		"arguments":   c.Dict("arguments", s.Schema{
+		}),
+		"messages":    c.Dict("message_stats", s.Schema{
+			"publish": s.Object{
+				"count": c.Int("publish"),
+				"details": c.Dict("publish_details", s.Schema{
+					"rate": c.Float("rate"),
+				}),
+			},
+			"publish_in": s.Object{
```

Comment:
We could make the above `publish.total.count` and this one here `publish.in` and the following `publish.out`. Is it correct the the above is just the total?

Reply:
The advantage of keeping it as `publish_in` is that all the groups of metrics under `messages` have the same structure, and they are also kept as a more direct mapping of source metrics.

Reply:
I'm always in favor or having all the related metrics under on group but it's really hard to have a clear rule here. I'm good with both.

Reply:
TBH I am not sure what "publish" is supposed to be. The [official documentation](https://rawcdn.githack.com/rabbitmq/rabbitmq-management/rabbitmq_v3_6_11/priv/www/doc/stats.html) says "Count of messages published.", but in my environment it is always 0, although publish_in and publish_out are both 0+.
Regarding the naming - it is up to you as I am not familiar with the metric set naming guidelines.